too-many-hooks/src/useFlexCorners.ts

```diff
@@ -0,0 +1,345 @@
+     *
+     * @type {Corner<T>}
+     */
+    right: Corner<T>
+  }
+  /**
+   * Information about elements contained in the rightmost column
+   *
+   * @type {{
+   *  top: Corner<T>
+   *  bottom: Corner<T>
+   * }}
+   */
+  right: {
+    /**
+     * The topmost element in the rightmost column
+     *
+     * @type {Corner<T>}
+     */
+    top: Corner<T>
+    /**
+     * The bottommost element in the rightmost column
+     *
+     * @type {Corner<T>}
+     */
+    bottom: Corner<T>
+  }
+  /**
+   * Information about elements contained in the bottommost row
+   *
+   * @type {{
+   *  left: Corner<T>
+   *  right: Corner<T>
+   * }}
+   */
+  bottom: {
+    /**
+     * The leftmost element in the bottommost row
+     *
+     * @type {Corner<T>}
+     */
+    left: Corner<T>
+    /**
+     * The rightmost element in the bottommost row
+     *
+     * @type {Corner<T>}
+     */
+    right: Corner<T>
+  }
+  /**
+   * Information about elements contained in the leftmost column
+   *
+   * @type {{
+   *  top: Corner<T>
+   *  bottom: Corner<T>
+   * }}
+   */
+  left: {
+    /**
+     * The topmost element in the leftmost column
+     *
+     * @type {Corner<T>}
+     */
+    top: Corner<T>
+    /**
+     * The bottommost element in the leftmost column
+     *
+     * @type {Corner<T>}
+     */
+    bottom: Corner<T>
+  }
+}
+
+/**
+ * Return values of the `useFlexCorners` hook
+ *
+ * @export
+ * @typedef {UseFlexCornersReturn}
+ */
+export interface UseFlexCornersReturn<T extends HTMLElement> {
+  /**
+   * Callback ref updater for tracking included elements
+   *
+   * @example
+   * To get corner for an array of strings placed as `div`s
+   * ```tsx
+   * <div style={{display: "flex", flexWrap: "wrap"}}>
+   *   {
+   *     strings.map((s, i) => <div ref={element => setElement(i, element)}>{s}</div>)
+   *   }
+   * </div>
+   * ```   *
+   * @type {(index: number, element: T) => void}
+   */
+  setElement: (index: number, element: T) => void
+  /**
+   * Information about elements in corner positions
+   *
+   * @type {Corners<T>}
+   */
+  corners: Corners<T> | null
+  /**
+   * Function to recalculate corners on a resize other than the standard window event
+   *
+   * @type {() => void}
+   */
+  onResize: () => void
+}
+
+/**
+ * `useFlexCorners` hook type
+ *
+ * @export
+ * @typedef {UseFlexCorners}
+ * @template T extends HTMLElement
+ */
+export type UseFlexCorners<T extends HTMLElement> = () => UseFlexCornersReturn<T>
+
+/**
+ * Dynamically locates corner elements of elements positioned with wrapping flex
+ *
+ * @implements {UseFlexCorners}
+ * @template T extends HTMLElement
+ * @returns {UseFlexCornersReturn<T>}
+ */
+const useFlexCorners = <T extends HTMLElement>(): UseFlexCornersReturn<T> => {
+  const [elements, { updateAt: setElement }] = useArray<T>([])
+
+  const mutationRecord = useMutationObserver(document, {
+    attributes: true,
+    subtree: true,
+  })
+
+  const findCorners = useCallback(() => {
```

Comment:
Might need to wrap the logic in this callback with a check that `document` is defined since that's usually not case on the initial render when using SSR.